packages/libs/lambda-at-edge/src/routing/redirector.ts

```diff
@@ -0,0 +1,108 @@
+import { compileDestination, matchPath } from "./matcher";
+import { RedirectData, RoutesManifest } from "../../types";
+import * as http from "http";
+
+/**
+ * Whether this is the default trailing slash redirect.
+ * This should only be used during build step to remove unneeded redirect paths.
+ * @param redirect
+ * @param basePath
+ */
+export function isTrailingSlashRedirect(
+  redirect: RedirectData,
+  basePath: string
+) {
+  if (basePath !== "") {
+    return (
+      redirect.statusCode === 308 &&
+      ((redirect.source === `${basePath}` &&
+        redirect.destination === `${basePath}/`) ||
+        (redirect.source === `${basePath}/` &&
+          redirect.destination === `${basePath}`) ||
+        (redirect.source === `${basePath}/:path+/` &&
+          redirect.destination === `${basePath}/:path+`) ||
+        (redirect.source === `${basePath}/:file((?:[^/]+/)*[^/]+\\.\\w+)/` &&
+          redirect.destination === `${basePath}/:file`) ||
+        (redirect.source === `${basePath}/:notfile((?:[^/]+/)*[^/\\.]+)` &&
+          redirect.destination === `${basePath}/:notfile/`))
+    );
+  } else {
+    return (
+      redirect.statusCode === 308 &&
+      ((redirect.source === "/:path+/" && redirect.destination === "/:path+") ||
+        (redirect.source === "/:path+" &&
+          redirect.destination === "/:path+/") ||
+        (redirect.source === "/:file((?:[^/]+/)*[^/]+\\.\\w+)/" &&
+          redirect.destination === "/:file") ||
+        (redirect.source === "/:notfile((?:[^/]+/)*[^/\\.]+)" &&
+          redirect.destination === "/:notfile/"))
+    );
+  }
+}
+
+/**
+ * Get the redirect of the given path, if it exists. Otherwise return null.
+ * @param path
+ * @param routesManifest
+ */
+export function getRedirectPath(
+  path: string,
+  routesManifest: RoutesManifest
+): { redirectPath: string; statusCode: number } | null {
+  const redirects: RedirectData[] = routesManifest.redirects;
+
+  for (const redirect of redirects) {
```

Comment:
Not one for now, but might be worth looking into trie router implementations out there which would scale better than this. There is an example impl. here https://github.com/pillarjs/routington

Reply:
Sure, we can look into that to improve routing in general